src/rdkafka_broker.c

```diff
@@ -4100,8 +4100,7 @@ static int rd_kafka_toppar_producer_serve(rd_kafka_broker_t *rkb,
 
                 /* Limit the number of in-flight requests (per partition)
                  * to the broker's sequence de-duplication window. */
-                max_requests = RD_MIN(max_requests,
-                                      RD_KAFKA_IDEMP_MAX_INFLIGHT - inflight);
```

Comment:
The `inflight` variable holds the number of inflight messages (`rd_atomic32_get(&rktp->rktp_msgs_inflight)`) instead of the number of inflight requests.

Reply:
Thanks @marcin-krystianc for spotting this! I'll check if other changes are needed but it doesn't seem at the moment.